src/app/entities/courses/store/effects/courses.effects.ts

```diff
@@ -0,0 +1,84 @@
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+
+import { Actions, createEffect, Effect, ofType } from '@ngrx/effects';
+
+import { of, throwError } from 'rxjs';
+import { catchError, map, switchMap, tap } from 'rxjs/operators';
+
+import { createCourse, deleteCourse, deleteCourseSuccess, loadCourses, loadCoursesPage, resetCourses, setCourses, updateCourse } from '@app/entities/courses/store/actions/courses.actions';
+import { AppRoutePath, ErrorMessage } from '@app/enums';
+import { CoursesService } from '@app/services';
+
+@Injectable()
+export class CoursesEffects {
+
+  constructor(private actions$: Actions, private router: Router, private coursesService: CoursesService) { }
+
+  @Effect()
+  loadCourses$ = this.actions$.pipe(
+    ofType(loadCourses),
+    switchMap(({ start, count, sort, textFragment }) => this.coursesService.loadCoursesPage({ start, count, sort, textFragment })),
+    catchError(({ error }) => {
+      alert(`${ErrorMessage.CoursesGet}: ${error}`);
+
+      return of([]);
+    }),
+    map(courses => resetCourses(courses)),
+  );
+
+  @Effect()
+  loadCoursesPage$ = this.actions$.pipe(
+    ofType(loadCoursesPage),
+    switchMap(({ start, count, sort, textFragment }) => this.coursesService.loadCoursesPage({ start, count, sort, textFragment })),
+    catchError(({ error }) => {
+      alert(`${ErrorMessage.CoursesGet}: ${error}`);
+
+      return of([]);
+    }),
+    map(courses => setCourses(courses)),
+  );
+
+  deleteCourse$ = createEffect(() => {
+    let courseId = null;
+
+    return this.actions$.pipe(
+      ofType(deleteCourse),
+      switchMap(({ id }) => {
+        courseId = id;
+
+        return this.coursesService.delete(courseId);
```

Comment:
it is a good practice to catchError where it can occur:

```
return this.actions$.pipe(
      ofType(deleteCourse),
      switchMap(({ id }) => { // and no longer need to create an additional variable
        return this.coursesService.delete(id).pipe(
catchError(({ error }) => {
        alert(`${ErrorMessage.CourseDelete}: ${error}`);

        return throwError(error); // this can be a separate action deleteCourseFail
      }),
      map(() => deleteCourseSuccess({ id })),
);
      }),
    
    );
```

Reply:
similar in other places

Reply:
Ok, thanks